doctor.py

```diff
@@ -43,21 +99,9 @@ def main(context: Dict[str, str], argv: Sequence[str] | None = None) -> int:
     executor = ThreadPoolExecutor(max_workers=len(checks))
     print(f"Running checks: {', '.join(f'{c.name}' for c in checks)}")
 
-    futures = {}
-    results = {}
-    for check in checks:
-        futures[check] = executor.submit(check.check)
-    for check, future in futures.items():
-        results[check] = future.result()
+    results = run_checks(checks, executor)
 
-    retry = []
-    for check, result in results.items():
-        ok, msg = result
-        if ok:
-            print(f"\t✅ check: {check.name}".expandtabs(4))
-            continue
-        print(f"\t❌ check: {check.name}{msg}".expandtabs(4))
-        retry.append(check)
+    retry = filter_failing_checks(results)
```

Comment:
NIT: Can we change this to failing_checks instead of retry?

retry to me sounds like checks that failed and we want to retry them, which isn't the case here.

Reply:
Whoops, yep!